### ISSUE TITLE
off_policy_td: gamma outside [0,1] silently accepted and applied

### ISSUE THREAD

**Issue by Svector-anu** [SlopDotCash/asi#1215]
`OffPolicyTDLinearLearner`, `ETDLinearLearner`, and `GradientTDLinearLearner` in `alberta_framework/core/off_policy_td.py` validate the per-transition discount `gamma` with `jnp.isfinite(gamma_s)` only. Any finite value is accepted and applied, including negative discounts or discounts greater than 1.

Reproduction (current tree):

```text
OffPolicyTDLinearLearner -0.25 applied= True step= 1
OffPolicyTDLinearLearner 1.25 applied= True step= 1
ETDLinearLearner -0.25 applied= True step= 1
ETDLinearLearner 1.25 applied= True step= 1
GradientTDLinearLearner -0.25 applied= True step= 1
GradientTDLinearLearner 1.25 applied= True step= 1
```

A negative `gamma` flips the sign contribution of the bootstrapped next-state value in the TD target; `gamma > 1` causes eligibility traces (`decay = gamma * lambda`) and the TD target's bootstrap term to grow rather than decay over repeated updates, instead of converging. Both silently corrupt learning instead of failing the transaction closed, the same way every other malformed-input path in this update already does (see `jnp.isfinite` checks on `reward`, `rho`, `observation` in the same predicate).

Same bug class as `swift_td.py::SwiftTD.update` (fixed in #1110): an out-of-range TD discount factor silently accepted and applied rather than rejected, in a sibling off-policy TD implementation.

`OffPolicyTDLinearLearner` and `GradientTDLinearLearner` are exported from both `alberta_framework/core/__init__.py` and the top-level `alberta_framework/__init__.py`; `run_gradient_td_learning_loop` passes caller-supplied `gammas` arrays straight into `GradientTDLinearLearner._update_jit` via `jax.lax.scan`. `ETDLinearLearner` is public from its own module but not top-level re-exported.

Fix incoming in a linked PR: require `0 <= gamma <= 1` in the existing finite-input gate (IEEE comparisons already reject NaN, so the prior `isfinite` check becomes redundant once this lands).
